imagescan/controller_test.go

```diff
@@ -643,6 +648,191 @@ func TestSubscriber(t *testing.T) {
 	})
 }
 
+func TestController_findBestNodeAndMode(t *testing.T) {
+	log := logrus.New()
+	log.SetLevel(logrus.DebugLevel)
+
+	t.Run("fallbacks when img had error", func(t *testing.T) {
+		cfg := config.ImageScan{
+			Mode:          string(imgcollectorconfig.ModeHostFS),
+			CPURequest:    "1",
+			MemoryRequest: "100Mi",
+		}
+
+		resMem := resource.MustParse("500Mi")
+		resCpu := resource.MustParse("2")
+
+		lessResMem := resource.MustParse("400Mi")
+		lessResCpu := resource.MustParse("1")
+
+		controller := newTestController(log, cfg)
+		controller.delta.nodes = map[string]*node{
+			"node1": {
+				name:           "node1",
+				architecture:   "amd64",
+				allocatableMem: resMem.AsDec(),
+				allocatableCPU: resCpu.AsDec(),
+				castaiManaged:  true,
+			},
+			"node2": {
+				name:           "node2",
+				architecture:   "amd64",
+				allocatableMem: lessResMem.AsDec(),
+				allocatableCPU: lessResCpu.AsDec(),
+				castaiManaged:  true,
+			},
+		}
+
+		img := &image{
+			key: "img1amd64img",
+			nodes: map[string]*imageNode{
+				"node1": {},
+				"node2": {},
+			},
+			lastScanErr: errImageScanLayerNotFound,
+		}
+
+		node, mode, err := controller.findBestNodeAndMode(img)
+		assert.NoError(t, err)
```

Comment:
Nitpic: Would be nice to have same style with `r := assert.New(t)` as in all tests.